### PR TITLE
feature(ui-ux): create - recovery words screen

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      style={tailwind('flex-1')}
+    >
+      <View style={tailwind('flex flex-row justify-center')}>
+        <Image
+          style={{ width: 200, height: 136 }}
+          source={isLight ? LightNewWallet : DarkNewWallet}
+        />
+      </View>
+      <ThemedText
+        style={tailwind('mt-7 text-base text-center')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:53]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        light={tailwind('text-mono-light-v2-900')}
+        dark={tailwind('text-mono-dark-v2-900')}
+      >
+        {translate('screens/Guidelines', 'You will be shown 24 recovery words on the next screen. Keep your 24-word recovery safe as it will allow you to recover access to the wallet')}
+      </ThemedText>
+      <ThemedText
+        style={tailwind('mt-2 text-base text-center font-semibold')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:60]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          <Checkbox
+            value={isEnabled}
+            onValueChange={toggleSwitch}
+            style={tailwind('h-6 w-6 mt-1 rounded')}
+            color={isEnabled ? getColor('brand-v2-500') : undefined}
+          />
+          <ThemedText
+            light={tailwind('text-mono-light-v2-700')}
+            dark={tailwind('text-mono-dark-v2-700')}
+            style={tailwind('text-xs ml-4')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:104]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                dark={tailwind('text-mono-dark-v2-100')}
+                iconType='Feather'
+                name={g.icon}
+                size={14}
+              />
+            </ThemedView>
+            <ThemedText
+              light={tailwind('text-mono-light-v2-700')}
+              dark={tailwind('text-mono-dark-v2-700')}
+              style={tailwind('text-xs ml-4')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:88]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            >
+              {translate('screens/Guidelines', g.title)}
+            </ThemedText>
+          </View>
+       ))}
+        <View style={tailwind('flex-row items-start mt-3')}>
+          <Checkbox
+            value={isEnabled}
+            onValueChange={toggleSwitch}
+            style={tailwind('h-6 w-6 mt-1 rounded')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:98]: not sure if possible to change the checkmark color based on theme, currently it is always white, but figma has black in dark mode 
image

File: mobile-app/app/components/CreateWalletStepIndicatorV2.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          step={1}
+        />
+        {following()}
+      </View>
+    </View>
+  )
+}
+
+function getStepNodeStyle (isLight: boolean, current: number, step: number): { stepperStyle: string, textStyle: string } {
+  let stepperStyle
```
Comment (kyleleow) [inline mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:87]: consider assigning one of the if-else condition as the initial value?

File: mobile-app/app/components/CreateWalletStepIndicatorV2.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        {props.current > props.step
+        ? <ThemedIcon
+            size={18}
+            name='check'
+            iconType='Feather'
+            dark={tailwind('text-mono-dark-v2-00')}
+            light={tailwind('text-mono-light-v2-00')}
+          />
+        : (
+          <Text style={tailwind(`${textStyle} font-semibold text-lg absolute`)}>
```
Comment (kyleleow) [inline mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:124]: use v2 font

File: mobile-app/app/components/CreateWalletStepIndicatorV2.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    </View>
+  )
+}
+
+function Description (props: {content: string, textStyle: string }): JSX.Element {
+  return (
+    <ThemedText
+      dark={tailwind(props.textStyle)}
+      light={tailwind(props.textStyle)}
+      style={tailwind('text-center text-xs font-normal top-12 absolute w-20')}
```
Comment (kyleleow) [inline mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:142]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx (hunk 1 above)
Comment (kyleleow) [reply mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:53]: and add 14px of spacing at the side

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx (hunk 5 above)
Comment (fullstackninja864) [reply mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:98]: Currently There is no option to change checkmark color based on theme

Comment (kyleleow) [reply mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:98]: can you check with design team and see if they are with this?

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+          source={isLight ? LightNewWallet : DarkNewWallet}
+        />
+      </View>
+      <ThemedText
+        style={tailwind('mt-7 px-3 text-base font-normal-v2 text-center')}
+      >
+        {translate('screens/Guidelines', 'You will be shown 24 recovery words on the next screen. Keep your 24-word recovery safe as it will allow you to recover access to the wallet')}
+      </ThemedText>
+      <ThemedText
+        style={tailwind('mt-2 px-3 text-base text-center font-semibold-v2')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:56]:
```suggestion
        style={tailwind('mt-2 px-3 text-sm text-center font-semibold-v2')}
```

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateMnemonicWalletV2.tsx [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: (): JSX.Element => (
+        <TouchableOpacity
+          onPress={refreshRecoveryWords}
+          testID='reset_recovery_word_button'
+        >
+          <ThemedIcon
+            dark={tailwind('text-darkprimary-500')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateMnemonicWalletV2.tsx:56]: check the color again on figma

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx (hunk 5 above)
Comment (fullstackninja864) [reply mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateWalletGuidelinesV2.tsx:99]: Confirmed with design team, will be using white tick

File: mobile-app/app/components/CreateWalletStepIndicatorV2.tsx [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+          key={i * 2 + 1}
+          step={i + 1}
+        />)
+    }
+    return arr
+  }
+
+  return (
+    <View style={[tailwind('flex-col justify-center items-center w-full'), containerViewStyle]}>
+      <View style={tailwind('flex-row justify-center w-9/12 h-16')}>
```
Comment (kyleleow) [inline mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:73]: let's increase the width slightly, looks better
```suggestion
      <View style={tailwind('flex-row justify-center w-10/12 h-16')}>
```

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateMnemonicWalletV2.tsx [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+            }, 1000)
+          }
+        }
+      ]
+    })
+  }
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: (): JSX.Element => (
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateMnemonicWalletV2.tsx:50]: can we see how to add more right spacing? To align with the header left spacing

File: mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateMnemonicWalletV2.tsx [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+        <TouchableOpacity
+          onPress={refreshRecoveryWords}
+          style={tailwind('mr-3')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/CreateWallet/CreateMnemonicWalletV2.tsx:53]: after checking on ios, android and web, i think we can remove `mr-3` here

File: mobile-app/app/components/skeletonLoaders/MnemonicWordSkeletonLoaderV2.tsx [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+      style={tailwind(['py-5 mx-5 flex-row justify-center', { 'border-b': props.border }])}
+    >
+      <ContentLoader
+        speed={2}
+        width='100%'
+        height={20}
+        viewBox='0 0 328 20'
+        preserveAspectRatio='xMidYMid slice'
+        backgroundColor={getColor(isLight ? 'mono-light-v2-100' : 'mono-dark-v2-100')}
+        foregroundColor={isLight ? '#ffffff' : '#4a4a4a'}
```
Comment (kyleleow) [inline mobile-app/app/components/skeletonLoaders/MnemonicWordSkeletonLoaderV2.tsx:23]: from figma, should be `mono-light/dark-v2-00`

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/components/CreateWalletStepIndicatorV2.tsx [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+ */
+export function CreateWalletStepIndicatorV2 (props: StepIndicatorProps): JSX.Element {
+  const { isLight } = useThemeContext()
+  const {
+    current,
+    total,
+    style: containerViewStyle,
+    steps = []
+  } = props
+  if (total === undefined && steps.length === 0) {
```
[the remark this answers is not in the document]
Comment (fullstackninja864) [reply mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:41]: I also feel the same, updating the logic there

[the remark this answers is not in the document]
Comment (kyleleow) [reply mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:41]: returning `<></>` or simply `return` would not be to catch the error, I suggest to log it but without throwing error, so we can still be notified
```
Logging.error('Invalid prop for CreateWalletStepIndicator')
return 
```

[the remark this answers is not in the document]
Comment (kyleleow) [reply mobile-app/app/components/CreateWalletStepIndicatorV2.tsx:41]: the UI would probably look very ugly/cramped when you have more than 5 stepper

**Pull request merged**